writing-test-plans/test-plan-manifest.md

```diff
@@ -32,19 +32,89 @@ enabled = true
 go_version = "1.14"
 module_path = "github.com/your/module/name"
 exec_pkg = "."
+fresh_gomod = true
+modfile = "./go.mod" # Custom modfile
+
+#  GoProxyMode specifies one of "local", "direct", "remote".
+# 
+#    * The "local" mode (default) will start a proxy container (if one
+#      doesn't exist yet) with bridge networking, and will configure the
+#      build to use that proxy.
+#    * The "direct" mode sets the `GOPROXY=direct` env var on the go build.
+#    * The "remote" mode specifies a custom proxy. The `GoProxyURL` field
+#      must be non-empty.
+go_proxy_mode = "local"
+
+# GoProxyURL specifies the URL of the proxy when GoProxyMode = "custom".
+go_proxy_url = ""
+
+# RuntimeImage is the runtime image that the test plan binary will be
+# copied into. Defaults to busybox:1.31.1-glibc.
+runtime_image = "busybox:1.31.1-glibc"
+
+# BuildBaseImage is the base build image that the test plan binary will be
+# built from. Defaults to golang:1.16-buster
+build_base_image = "golang:1.16-buster"
+
+# SkipRuntimeImage allows you to skip putting the build output in a
+# slimmed-down runtime image. The build image will be emitted instead.
+skip_runtime_image = true
+
+# EnableGoBuildCache enables the creation of a go build cache and its usage.
+# When enabling for the first time, a cache image will be created with the
+# dependencies of the current plan state.
+# If this flag is unset or false, every build of a test plan will start
+# with a blank go container. If this flag is true, the builder will the last
+# cached image.
+enable_go_build_cache = false
+
+#  Cgo enables the creation of Go packages that call C code. By default it is disabled.
+#  Enabling CGO also enables dynamic linking. Disabling CGO (default) produces statically  linked binaries.
+#  If you ever see errors like the following, you are probably better off
+#  disabling CGO (and therefore enabling static linking).
+#    /testplan: error while loading shared libraries: libdl.so.2: cannot open shared object file: No such file or directory
+#  If you pass `true` to this flag, your test plan will be built with CGO_ENABLED=1
+enable_cgo = false
 
 [builders."exec:go"]
 enabled = true
 module_path = "github.com/your/module/name"
 
 [runners."local:docker"]
 enabled = true
+ulimits = [
+  "nofile=1048576:1048576",
+]
+
+[daemon]
+listen                    = ":8040"
+
+[daemon.scheduler]
+task_timeout_min          = 5
+task_repo_type            = "disk"
+
+[client]
+endpoint = "http://localhost:8040"
+user = "myname"
```

Comment:
Can you double-check all these options? It looks like we're passing daemon configuration in a test's manifest.

Reply:
Yes that was really confusing for me, because I thought that we can change daemon's and engine's configuration from test's manifest because I saw some manifests examples. I would remove all those engine's configuration from test-plan-manifest. What do you think? @laurentsenta

Reply:
Yes please, could you share the link where you've seen these options used in a test manifest?
This is definitely something we need to remove from wherever it is defined.

Also, in testground: `pkg/api/manifest.go` you'll find the spec for this file this will help filter the correct options.